modules/libcom/src/osi/osiSock.h

```diff
@@ -28,185 +34,367 @@ struct sockaddr;
 struct sockaddr_in;
 struct in_addr;
 
+/*!
+ * \brief Create a socket object
+ *
+ * Ask the operating system to create a socket and return its file descriptor.
+ *
+ * \param domain The socket domain, e.g. PF_INET
+ * \param type The type of socket, e.g. SOCK_STREAM (tcp) or SOCK_DGRAM (udp)
+ * \param protocol Typically unused and set to 0.
+ * \return A file descriptor referring to the socket, or -1 on error.
+ */
 LIBCOM_API SOCKET epicsStdCall epicsSocketCreate ( 
     int domain, int type, int protocol );
+
+/*!
+ * \brief Accept a connection on a listening stream socket.
+ *
+ * epicsSocketAccept is used with connection-based sockets.  It takes the next
+ * connection request in the queue and returns a new socket that can be used to
+ * communicate with the listener.
+ *
+ * \param sock socket with pending connections to accept
+ * \param pAddr If not NULL, this socket contains the address of the peer whose connection is being accepted
+ * \param[in,out] addrlen Length of the structure pointed to by @p pAddr, or NULL if \p pAddr is NULL.
```

Comment:
Is it possible to have some description as to why it is also an "out" parameter? It's not clear to me what info we get back in `addrlen` after calling the function.

Reply:
You're right, that could be clearer.  Here's my 2nd attempt:

 \param[in,out] addrlen Caller must initialize this to the length of the
 * structure pointed to by @p pAddr, or set to NULL if \p pAddr is NULL.  This
 * function will update the value of this parameter to the actual size of the
 * peer address.